psutil/_psaix.py

```diff
@@ -415,7 +410,12 @@ def exe(self):
 
     @wrap_exceptions
     def cmdline(self):
-        return self._proc_name_and_args()[1].split(' ')
+        try:
+            return cext.proc_args(self.pid) or ['']
+        except OSError as err:
+            if err.errno == errno.ESRCH:
+                raise NoSuchProcess(self.pid, self._name)
```

Comment:
i think it’s simpler if you just do this in C; you can call (from C) NoSuchProcess(“”) and return NULL

Reply:
The C `NoSuchProcess` just sets the exception to `OSError` with `ESRCH` (instead of setting the exception to psutil's `NoSuchProcess`) so it wouldn't change anything. However, I can probably remove this because the decorator `wrap_exception` catches this exception and does the translation.

Reply:
yeah

Reply:
yeah